README.md

```diff
@@ -283,3 +283,34 @@ Update appsettings.json:
   }
 }
 ```
+
+# Swagger Metadata
+
+If you want to publish the API metadata ([OpenAPI info object](https://swagger.io/specification/#info-object)) from the configured swaggers, specify a path from the list of configured paths using the `MetadataPath` option. This will overwrite the cluster metadata with that of the swagger metadata.
+
+Update appsettings.json:
+
+```json lines
+{
+  "ReverseProxy": {
+    "Clusters": {
+      "App1Cluster": {
+        "Destinations": {
+          "Default": {
+            "Address": "https://localhost:5101",
+            "Swaggers": [
+              {
+                "PrefixPath": "/proxy-app1",
+                "MetadataPath": "/swagger/v1/swagger.json", // <-- this line
```

Comment:
Maybe you need copy metadata from all paths instead of only one because it will be difficult choice. So add bool flag "CopyMetadata" to copy all data.

Reply:
I had the same idea too, but I don't think there is a good way to copy metadata from all of the swagger paths.

Let me explain my thought process.

In the [`ConfigureSwaggerOptions.cs`](https://github.com/andreytreyt/yarp-swagger/blob/main/sample/Yarp/Configs/ConfigureSwaggerOptions.cs), we're creating a single swagger document per cluster in the [`appsettings.json`](https://github.com/andreytreyt/yarp-swagger/blob/main/sample/Yarp/appsettings.json):
```csharp
foreach (var cluster in _reverseProxyDocumentFilterConfig.Clusters)
{
    options.SwaggerDoc(cluster.Key, new OpenApiInfo {Title = cluster.Key, Version = cluster.Key});
}
```
**In other words, for each configured cluster, we have a single OpenAPI spec document.**

In the OpenAPI spec, there is only a single required [Info object](https://swagger.io/specification/#info-object) that describes the metadata for the spec document. In the event there are multiple swagger paths configured, that means the cluster has several OpenAPI specs. Each OpenAPI spec has its own `Info` object, so there are several `Info` objects we could copy. 

**But our dilemma is our cluster's OpenAPI spec can only accept a _single_ `Info` object.**

We could try to somehow merge all of the `Info` objects together into a single `Info` object, but I don't think that makes sense. Trying to combine fields like `title`, `summary`, or `version` from multiple API specs could create non-sensical metadata.

So, I think we should allow the user to pick which OpenAPI spec document we should copy the `Info` object from. In the event they have more than one OpenAPI spec, they should just manually create the `Info` object like they do today.

To be honest, this feature would be really helpful for my own use case. I generally only have 1 OpenAPI spec per cluster and I suspect that is true for most people's use cases. That's why I think this should be an optional feature that people can opt into.

Reply:
Okay! Thanks a lot for your complete answer.